tss/implementation.go

```
@@ -165,191 +150,110 @@ func (t *Engine) beginTSSSign(vaaDigest []byte, chainID vaa.ChainID, consistency
 	}
 
-	// waiting for the reply.
-	sigPrepInfo, err := outOfChannelOrDone(t.ctx, cmd.reply)
-	if err != nil {
-		return sigPreparationInfo{}, fmt.Errorf("failed to get inactive guardians: %w", err)
+	if err := validateTrackingID(info.TrackingID); err != nil {
+		return err
 	}
 
-	return sigPrepInfo, nil
-}
+	t.createSignatureMetrics(info.TrackingID)
 
-// prepareThenAnounceNewDigest updates the inner state of the engine before announcing to others about a new digest seen.
-func (t *Engine) prepareThenAnounceNewDigest(sigtask party.SigningTask, consistencyLvl uint8, mt signingMeta) error {
-	signinginfo, err := t.fp.GetSigningInfo(sigtask)
+	info, err = t.fp.AsyncRequestNewSignature(sigtask)
 	if err != nil {
-		return fmt.Errorf("couldnt generate signing task: %w", err)
+		return err
 	}
 
-	sgCmd := &signCommand{
-		SigningInfo:       signinginfo,
-		passedToFP:        false, // set to true only after FP actually received the message.
-		digestconsistancy: consistencyLvl,
-		signingMeta:       mt,
-	}
+	t.logger.Info(
+		"guardian started signing protocol",
 
-	if err := intoChannelOrDone[ftCommand](t.ctx, t.ftCommandChan, sgCmd); err != nil {
-		return fmt.Errorf("couldn't inform the tracker of the signature start: %w", err)
-	}
+		zap.String("trackingID", info.TrackingID.ToString()),
+		// zap.String("ChainID", chainID.String()),
+		zap.Any("committee", t.getCommitteeNetworkNames(info.SigningCommittee)),
+	)
 
 	return nil
 }
 
-func (t *Engine) getProtocolForChain(chainID vaa.ChainID) common.ProtocolType {
-	for _, ecdsaChain := range t.GuardianStorage.EcdsaChains {
-		if ecdsaChain == chainID {
-			return common.ProtocolECDSASign
+func (t *Engine) getCommitteeNetworkNames(pids []*common.PartyID) []string {
+	ids := make([]string, 0, len(pids))
+	for _, pid := range pids {
+		id, err := t.GuardianStorage.fetchIdentityFromPartyID(pid)
+		if err != nil {
+			t.logger.Warn("couldn't find identity for partyID", zap.Any("partyID", pid))
```

I think it might be better to return an error in this case, though not critical (if instead there's a check that the length of the returned array matches that of pid)